### DISCUSSION TITLE
When to use --shamefully-hoist flag

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [pnpm/pnpm#5417] [Q&A] [answered]
I recently moved to pnpm.

When trying it on Nuxt 3, I was told to use the --shamefully-hoist flag.
I almost turn back to npm, but because of pnpm's surprising speed, 
I decided to let it pass and read more about that flag.

But where I'm lost is about when and when not to use that flag.

And maybe creating a list of libraries/frameworks (if has not been made already?) specifying which of them need that flag when installing their dependencies.

Awesome manager, thanks to the team and the creator(s).

**Comment by angelhdzmultimedia** [comment]
Closing this issue. I wish there were a list of packages that require the flag.
At least Nuxt now adds a .npmrc file with the flag set to true in a new project.
I hope other packages that need it do this as well, so we don't have to worry about if we need the flag or not to avoid issues with peer dependencies and stuff.

**Reply by zkochan** [reply, marked as answer]
Yarn maintains a compatibility DB: https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-extensions/sources/index.ts

It patches some of the known broken packages in the ecosystem. pnpm also uses this DB to patch deps. If you know about some broken package, you can update that DB.

**Reply by angelhdzmultimedia** [reply]
Thanks @zkochan ! pnpm is awesome! Wishing you and the team a lot of health, peace, and success!
